packages/jest/src/utils.js

```diff
@@ -58,7 +58,12 @@ function getClassNameProp(node) {
   return (node && node.prop('className')) || ''
 }
 
-function getClassNamesFromEnzyme(selectors, node) {
+function getClassNamesFromEnzyme(selectors, nodeWithFragment) {
+  // TODO: Does approach work with older Emotion versions? Does it cause false positives with components named "Fragment" that aren't `React.Fragment`?
+  const node =
+    nodeWithFragment.name() === 'Fragment'
+      ? nodeWithFragment.children()
+      : nodeWithFragment
```

Comment:
I'll take a look into this issue and resolve this TODO comment before merging this

Reply:
@eps1lon i've tweaked this and left the comment about the chosen solution, what do you think - is relying on `Symbol.for` OK here or should we add `react-is` as a dependency and use that?

Reply:
`react-is` comes with its own problems (e.g. if it's a dependency people might have a different version of React installed. See https://github.com/facebook/react/issues/20099 for more details).

I'd say pick the one that you're most comfortable with and see if it works for the scenarios that it's used with.

Reply:
thanks for the feedback, I've noticed later that we are actually already using `Symbol.for` for different checks, so I'm just going to stick with this strategy